### PR TITLE
Support 2016-07-01 version of resources API spec. 

### PR DESCRIPTION
**Pull request by TianoMS**

### CODE REVIEW

**Comment by azurecla** [comment]
Hi __@TianoMS__, I'm your friendly neighborhood Azure Pull Request Bot (You can call me AZPRBOT). Thanks for your contribution!
    <span>
        In order for us to evaluate and accept your PR, we ask that you sign a contribution license agreement. It's all electronic and will take just minutes. I promise there's no faxing. https://cla.azure.com.
    </span>

TTYL, AZPRBOT;

File: arm-resources/resources/2016-07-01/swagger/resources.json [hunk 1]
```diff
@@ -0,0 +1,2449 @@
...
+          },
+          "description": "The paths for an alias."
+        }
+      }
+    },
+    "ZoneMappingType": {
+      "properties": {
+        "location": {
+          "type": "string",
+          "descripton": "The location of this zone mapping."
```
Comment (John-Hart) [inline arm-resources/resources/2016-07-01/swagger/resources.json]: description is spelled wrong

**Comment by John-Hart** [comment]
@TianoMS it would really help speed up future reviews of new versions if the first commit that you make is just an exact copy of the existing version in the new location and then follow that with additional commits for any changes you're making for the new version. This will really help identify what has changed.

File: arm-resources/resources/2016-07-01/swagger/resources.json [hunk 2]
```diff
@@ -0,0 +1,2449 @@
...
+            "in": "path",
+            "required": true,
+            "type": "string",
+            "description": "Resource identity."
+          },
+          {
+            "name": "api-version",
+            "in": "query",
+            "required": true,
+            "type": "string"
```
Comment (John-Hart) [inline arm-resources/resources/2016-07-01/swagger/resources.json]: Please consider adding a description element, it's essential for maintaining reference documentation.

File: arm-resources/resources/2016-07-01/swagger/resources.json [hunk 3]
```diff
@@ -0,0 +1,2449 @@
...
+          },
```
Comment (John-Hart) [inline arm-resources/resources/2016-07-01/swagger/resources.json]: Please consider adding a description element, it's essential for maintaining reference documentation.

File: arm-resources/resources/2016-07-01/swagger/resources.json [hunk 4]
```diff
@@ -0,0 +1,2449 @@
...
+            "in": "path",
+            "required": true,
+            "type": "string",
+            "description": "Resource identity."
+          },
+          {
+            "name": "api-version",
+            "in": "query",
+            "required": true,
+            "type": "string"
```
Comment (John-Hart) [inline arm-resources/resources/2016-07-01/swagger/resources.json]: Same as above, please consider adding a description

File: arm-resources/resources/2016-07-01/swagger/resources.json [hunk 5]
```diff
@@ -0,0 +1,2449 @@
...
+          },
```
Comment (John-Hart) [inline arm-resources/resources/2016-07-01/swagger/resources.json]: Same as above, please consider adding a description

File: arm-resources/resources/2016-07-01/swagger/resources.json [hunk 6]
```diff
@@ -0,0 +1,2449 @@
...
+    "Deployment": {
+      "properties": {
+        "properties": {
+          "$ref": "#/definitions/DeploymentProperties",
+          "description": "The deployment properties."
+        }
+      },
+      "description": "Deployment operation parameters."
+    },
+    "DeploymentExportResult": {
```
Comment (John-Hart) [inline arm-resources/resources/2016-07-01/swagger/resources.json]: Please consider adding a description element, it's essential for maintaining reference documentation.

File: arm-resources/resources/2016-07-01/swagger/resources.json [hunk 7]
```diff
@@ -0,0 +1,2449 @@
...
+          },
+          "description": "Validation error."
+        }
+      },
+      "required": [
+        "code",
+        "message"
+      ]
+    },
+    "AliasPathType": {
```
Comment (John-Hart) [inline arm-resources/resources/2016-07-01/swagger/resources.json]: Please consider adding a description element, it's essential for maintaining reference documentation.

File: arm-resources/resources/2016-07-01/swagger/resources.json [hunk 8]
```diff
@@ -0,0 +1,2449 @@
...
+        "apiVersions": {
+          "type": "array",
+          "items": {
+            "type": "string"
+          },
+          "description": "The api versions."
+        }
+      }
+    },
+    "AliasType": {
```
Comment (John-Hart) [inline arm-resources/resources/2016-07-01/swagger/resources.json]: Please consider adding a description element, it's essential for maintaining reference documentation.

File: arm-resources/resources/2016-07-01/swagger/resources.json [hunk 9]
```diff
@@ -0,0 +1,2449 @@
...
+        "paths": {
+          "type": "array",
+          "items": {
+            "$ref": "#/definitions/AliasPathType"
+          },
+          "description": "The paths for an alias."
+        }
+      }
+    },
+    "ZoneMappingType": {
```
Comment (John-Hart) [inline arm-resources/resources/2016-07-01/swagger/resources.json]: Please add a description element for ZoneMappingType, it's essential for maintaining reference documentation.

File: arm-resources/resources/2016-07-01/swagger/resources.json [hunk 10]
```diff
@@ -0,0 +1,2449 @@
...
+      "properties": {
+        "location": {
+          "type": "string",
+          "descripton": "The location of this zone mapping."
+        },
+        "zones": {
+          "type": "array",
+          "items": {
```
Comment (John-Hart) [inline arm-resources/resources/2016-07-01/swagger/resources.json]: Please add a description element for items.

File: arm-resources/resources/2016-07-01/swagger/resources.json [hunk 11]
```diff
@@ -0,0 +1,2449 @@
...
+    },
+    "ProviderResourceType": {
+      "properties": {
+        "resourceType": {
+          "type": "string",
+          "description": "The resource type."
+        },
+        "locations": {
+          "type": "array",
+          "items": {
```
Comment (John-Hart) [inline arm-resources/resources/2016-07-01/swagger/resources.json]: Please consider adding a description element.

File: arm-resources/resources/2016-07-01/swagger/resources.json [hunk 12]
```diff
@@ -0,0 +1,2449 @@
...
+        "aliases": {
+          "type": "array",
+          "items": {
+            "$ref": "#/definitions/AliasType"
+          },
+          "description": "The aliases that are supported by this resource type."
+        },
+        "apiVersions": {
+          "type": "array",
+          "items": {
```
Comment (John-Hart) [inline arm-resources/resources/2016-07-01/swagger/resources.json]: Please consider adding a description element.

File: arm-resources/resources/2016-07-01/swagger/resources.json [hunk 13]
```diff
@@ -0,0 +1,2449 @@
...
+        "zoneMappings": {
+          "type": "array",
+          "items": {
+            "$ref": "#/definitions/ZoneMappingType"
+          },
+          "description": "The zone mappings supported by this resource type."
+        },
+        "properties": {
+          "type": "object",
+          "additionalProperties": {
```
Comment (John-Hart) [inline arm-resources/resources/2016-07-01/swagger/resources.json]: Please consider adding a description element.

File: arm-resources/resources/2016-07-01/swagger/resources.json [hunk 14]
```diff
@@ -0,0 +1,2449 @@
...
+        "properties": {
+          "$ref": "#/definitions/ResourceGroupProperties"
+        },
+        "location": {
+          "type": "string",
+          "description": "The location of the resource group. It cannot be changed after the resource group has been created. Has to be one of the supported Azure Locations, such as West US, East US, West Europe, East Asia, etc."
+        },
+        "tags": {
+          "type": "object",
+          "additionalProperties": {
```
Comment (John-Hart) [inline arm-resources/resources/2016-07-01/swagger/resources.json]: Please consider adding a description element.

File: arm-resources/resources/2016-07-01/swagger/resources.json [hunk 15]
```diff
@@ -0,0 +1,2449 @@
...
+      "required": [
+        "nextLink"
+      ],
+      "description": "List of resource groups."
+    },
+    "ResourcesMoveInfo": {
+      "properties": {
+        "resources": {
+          "type": "array",
+          "items": {
```
Comment (John-Hart) [inline arm-resources/resources/2016-07-01/swagger/resources.json]: Please consider adding a description element.

File: arm-resources/resources/2016-07-01/swagger/resources.json [hunk 16]
```diff
@@ -0,0 +1,2449 @@
...
+          "description": "The target resource group."
+        }
+      },
+      "description": "Parameters of move resources."
+    },
+    "ExportTemplateRequest": {
+      "properties": {
+        "resources": {
+          "type": "array",
+          "items": {
```
Comment (John-Hart) [inline arm-resources/resources/2016-07-01/swagger/resources.json]: Please consider adding a description element.

File: arm-resources/resources/2016-07-01/swagger/resources.json [hunk 17]
```diff
@@ -0,0 +1,2449 @@
...
+        "count": {
+          "$ref": "#/definitions/TagCount",
+          "description": "The tag count."
+        },
+        "values": {
+          "type": "array",
+          "items": {
+            "$ref": "#/definitions/TagValue"
+          },
+          "description": "The list of ta
```
Comment (John-Hart) [inline arm-resources/resources/2016-07-01/swagger/resources.json]: Please consider adding a description element.

Comment (John-Hart) [inline arm-resources/resources/2016-07-01/swagger/resources.json]: Please consider adding a description element.

Comment (John-Hart) [inline arm-resources/resources/2016-07-01/swagger/resources.json]: Please consider adding a description element.

Comment (John-Hart) [inline arm-resources/resources/2016-07-01/swagger/resources.json]: Please consider adding a description element.

**Comment by TianoMS** [comment]
@John-Hart Thanks John for the advice. I'll improve the process next time. Also, I have addressed the other comments on descriptions.

File: arm-resources/resources/2016-07-01/swagger/resources.json [hunk 18]
```diff
@@ -0,0 +1,2459 @@
...
+    },
+    "ZoneMappingType": {
+      "properties": {
+        "location": {
+          "type": "string",
+          "description": "The location of this zone mapping."
+        },
+        "zones": {
+          "type": "array",
+          "items": {
```
Comment (John-Hart) [inline arm-resources/resources/2016-07-01/swagger/resources.json]: Please add a description for items

Comment (TianoMS) [inline arm-resources/resources/2016-07-01/swagger/resources.json]: Do we need it in "items"? As I see we have "description" for "zones" (save level as "items").

File: arm-resources/resources/2016-07-01/swagger/resources.json [hunk 19]
```diff
@@ -1684,7 +1689,8 @@
           },
           "description": "The api versions."
         }
-      }
+      },
+      "description": "The type of alias paths. "
```
Comment (John-Hart) [inline arm-resources/resources/2016-07-01/swagger/resources.json]: is it alias paths (plural) or just path without the "s".  

Would it be clearer with something like:
     "The type of the alias for the path"

File: arm-resources/resources/2016-07-01/swagger/resources.json (hunk 18 above)
Comment (John-Hart) [inline arm-resources/resources/2016-07-01/swagger/resources.json]: No you're right, as long as it's on the property.  I got carried away :-)

File: arm-resources/resources/2016-07-01/swagger/resources.json (hunk 19 above)
Comment (TianoMS) [inline arm-resources/resources/2016-07-01/swagger/resources.json]: It's alias paths (plural). Basically an alias has paths, and this is the type for each of the path. I can update to "the type of the paths for the alias" but not sure if it looks better.

Comment (John-Hart) [inline arm-resources/resources/2016-07-01/swagger/resources.json]: that reads better

**Comment by John-Hart** [comment]
LGTM

**Pull request merged**